### DISCUSSION TITLE
Some ljspeech model/vocoder pairs producing terrible quality

### DISCUSSION THREAD

**Discussion by mittimithai** [coqui-ai/TTS#741] [General Q&A]
**Describe the bug**
Mixing and matching different tts models and vocoders can result in very bad speech quality. The behavior isn't simple.

**To Reproduce**
1. Run the following command that uses only prebuilt models(from your own paths):
python ../../TTS/TTS/bin/synthesize.py -model_name tts_models/en/ljspeech/tacotron2-DDC  --vocoder_name vocoder_models/en/ljspeech/multiband-melgan  --out_path ../../workingtmp/tts2ddc_melgan_prebuilt.wav --text "The thick elm forest was nearly overwhelmed by Dutch Elm disease."

The quality seems very bad. 

2. The same command with hifi vocoder sounds great:
python ../../TTS/TTS/bin/synthesize.py --model_name tts_models/en/ljspeech/tacotron2-DDC  --vocoder_name vocoder_models/en/ljspeech/hifigan_v2  --out_path ../../workingtmp/tts2ddc_hifi_prebuilt.wav --text "The thick elm forest was nearly overwhelmed by Dutch Elm disease."

**Expected behavior**
Both speech samples should sound good quality since these are both released models.

**Environment (please complete the following information):**
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Ubuntu 18.04.5 LTS
- **PyTorch or TensorFlow version (use command below)**:1.9.0
- **Python version**:3.8.8
- **CUDA/cuDNN version**:NA
- **GPU model and memory**: NA
- **Exact command to reproduce**: python ../../TTS/TTS/bin/synthesize.py -model_name tts_models/en/ljspeech/tacotron2-DDC  --vocoder_name vocoder_models/en/ljspeech/multiband-melgan  --out_path ../../workingtmp/tts2ddc_melgan_prebuilt.wav --text "The thick elm forest was nearly overwhelmed by Dutch Elm disease."

**Additional context**
I simply presumed that there is something wrong with the older vocoders when hifi was working fine, but using melgan on my custom tts model (using tts_models/en/ljspeech/tacotron2-DDC as a template config on the LJSpeech dataset with no real changes) actually sounds good (not as good as prebuilt) and hifi is barely intelligible.  It feels like there is a hidden variable I am missing here.

**Comment by thorstenMueller** [comment]
Hello @mittimithai. Not every vocoder is compatible with every tts model. If audio parameters do not match the quality will not sound good at all. But that's not a bug in my opinion. And the quality of vocoders themself can differ. HifiGAN is producing better results than MelGAN, as you figured out yourself.

You do not need to pass an vocoder_name argument on the command line. The "best" vocoder model for each tts model is linked in .models.json file.

**Comment by mittimithai** [comment]
To be sure it isn't that hifi is merely better, it is night and day (wav file is barely intelligible) (attached). Can you confirm this is the delta that should be expected?

I would have just assumed that melgan was old/deprecated but in my hands an LJSpeech model I trained with the *same* params as "tacotron2-DDC" (I had to reduce batch size to 32 on my machine and turn off graduated schedule), the same hifi vocoder that sounds great with tacotron2-DDC sounds awful with my model while melgan sounds ok right now.   If you think this is all explained by audio processing params, so be it, but it feels like I might be missing something that isn't explicit.

I'd like to make sure that I am not doing something wrong with vocoding before I submit the PR for my changes for custom symbol embeddings.


[ljspeech_tt2ddc_hifi_melgan.zip](https://github.com/coqui-ai/TTS/files/6997572/ljspeech_tt2ddc_hifi_melgan.zip)

**Comment by erogol** [comment]
If something is terrible, there is something wrong. So Tacotron-DDC is not compatible with the MelGAN vocoder. 

If you are not experimenting and after the best result, just let it run with the default vocoder for each tts model.

If you don't specify the vocoder model, the tts model uses the default vocoder that is set for it.
